### PR TITLE
Daily crusher

### PR DESCRIPTION
**Pull request by anapecic**

### CODE REVIEW

File: components/DailyCrusher/DailyCrusher.js
Comment (eneben) [inline components/DailyCrusher/DailyCrusher.js:103]:
```suggestion
        {isCrusherDetailsVisible && (
          <>
            {dailyCrusher.exercises.map((exercise) => {
              return (
                <StyledExerciseSetsReps key={uid()}>
                  <StyledBoldText>{exercise.name}: </StyledBoldText> Sets:{" "}
                  {exercise.sets} / Reps: {exercise.reps}
                </StyledExerciseSetsReps>
              );
            })}
            <StyledDescriptionText>
              {dailyCrusher.description2}
            </StyledDescriptionText>
            <StyledDescriptionText>
              {dailyCrusher.description3}
            </StyledDescriptionText>
          </>
        )}
```
If you only have null after the : in the ternary operator, you can also use the shorter notation with &&. I think that's clearer, cleaner notation.

File: components/DailyCrusher/DailyCrusher.js
Comment (eneben) [inline components/DailyCrusher/DailyCrusher.js:109]: I don't understand why you have the double negation with `!isCrusherDetailsVisible : null ? ...`
Directly above that you have the {isCrusherDetailsVisible && ....} - why don't you just add the "show less" button there? I think that would be much more logical. 🤔

File: components/DailyCrusher/DailyCrusher.js
Comment (eneben) [inline components/DailyCrusher/DailyCrusher.js:129]: I think rem is great for font sizes etc., but illogical for spacing such as padding. If I imagine that I have a visual impairment and my default font size is very large, then I want to be able to read the font well, but I don't want to have much larger spacing – otherwise I might not be able to read much on a small screen. The design might be a little worse, but accessibility will be better, and that's what's important when using rem etc.

File: components/DailyCrusher/DailyCrusher.js
Comment (eneben) [inline components/DailyCrusher/DailyCrusher.js:165]: For centering horizontally I like to use `margin: 0 auto` or `left: 0; right: 0;`

File: components/DailyCrusher/DailyCrusher.js
Comment (eneben) [inline components/DailyCrusher/DailyCrusher.js:121]: We were told in the course that we should not use any margin at all if possible. So we try to put containers around it and give them padding when necessary.

File: components/Workout/Workout.js
Comment (eneben) [inline components/Workout/Workout.js:65]:
```suggestion
            <ModalContent onClick={(event) => event.stopPropagation()}>
```
We were told by the coaches please not to use these abbreviations.

File: components/Workout/Workout.js
Comment (eneben) [inline components/Workout/Workout.js:74]:
```suggestion
        ) : null}
```
see above. you could use `{deleteMode && (...)}` withour `:null`

File: components/Workout/Workout.js
Comment (eneben) [inline components/Workout/Workout.js:75]:
```suggestion
        {!spotlightMode && (
```
Honestly, I find it strange when the code behind the ? in the ternary is null. I understand that you only want something to be displayed if the condition is not true. So why don't you use the NOT operator `!` and `&&`? I personally find that much easier to read, but it doesn't change your logic. 🤷‍♀️

File: components/Workout/Workout.js
Comment (eneben) [inline components/Workout/Workout.js:92]: see above
```suggestion
        {!spotlightMode && (
```

File: lib/workouts.js
Comment (eneben) [inline lib/workouts.js:1]: Wow, a lot of content! We had Chat GPT generate more content for our project so that the app would be a bit more interesting. :wink:

File: pages/index.js
Comment (eneben) [inline pages/index.js:69]: I think this feature is really cool! Just a small tip: The emoji in the span element is already good. If you now give the span element an aria label, that's great for accessibility!
```suggestion
              <span aria-label="sad emoji">😔</span>
```

File: pages/index.js
Comment (eneben) [inline pages/index.js:77]: see above
```suggestion
              <span aria-label="super happy emoji">🤩</span>
```

File: styledComponents.js
Comment (eneben) [inline styledComponents.js:120]: I would try to be consistent with the color types. Here you use RGB once and hex once and further down you write "black" as a word. :wink:

**Review by eneben — CHANGES_REQUESTED** [review]
First of all: I think your code is good. I've written quite a few comments for you, but they're all minor details.

Unfortunately, I still can't approve the PR, sorry: If I see it correctly, one acceptance criterion is not met, namely "Clicking the workout card should redirect to the full workout details page."

Your link always leads to the workouts page, but not to the specific workout and certainly not to the detailed view. I always end up at the top of the workout page. I'm sure that can be fixed somehow with scrollIntoView or something. The detailed view should also open up. I'm sure you can add that.

Otherwise, great work, your app looks great!

File: components/DailyCrusher/DailyCrusher.js
Comment (anapecic) [reply components/DailyCrusher/DailyCrusher.js:103]: We as a group decided that we want to use the ternary operators for all conditional renderings :)

File: components/DailyCrusher/DailyCrusher.js
Comment (anapecic) [reply components/DailyCrusher/DailyCrusher.js:109]: The double negation was just my laziness, haha. I kept the ternary separate from the other though, since I wanted to move it to the bottom of the article and stand by itself.

File: components/DailyCrusher/DailyCrusher.js
Comment (anapecic) [reply components/DailyCrusher/DailyCrusher.js:129]: We prefer rem as our unit for most of the cases. We also tested out what would happen if we increase the global font size to see what happens in terms of the padding and are fine with how it looks. Good point though, will keep that in mind :)

File: components/DailyCrusher/DailyCrusher.js
Comment (anapecic) [reply components/DailyCrusher/DailyCrusher.js:165]: This does not work here since we are working with position: absolute, meaning here it is necessary to work with top/left etc. If we were to use left: 0 & right: 0, the button would span across the div, which is not the look we are looking for :)

File: components/DailyCrusher/DailyCrusher.js
Comment (anapecic) [reply components/DailyCrusher/DailyCrusher.js:121]: Thank you for this advice :)

File: components/DailyCrusher/DailyCrusher.js
Comment (eneben) [inline components/DailyCrusher/DailyCrusher.js:66]: Hallo noch mal :)
Wenn es Absicht ist, dass Ihr nur auf die Workoutliste verlinkt (wie bei Slack geschrieben), gebe ich euch den grünen Haken natürlich. Ich find's aber ehrlich gesagt unlogisch und würde das an Eurer Stelle noch mal überdenken. Ich hab eben kurz Chat GPT gefragt: Eigentlich scheint das so zu gehen, wie ich gedacht habe, auch ohne dass Ihr auf eine extra Seite verweist. Vielleicht hilft euch das ja:
https://chatgpt.com/share/66ea9dd5-6978-8013-8ec8-fac0bff078e0
Viel Spaß weiterhin!

**Review by eneben — APPROVED** [review]
Hab Euch einen neuen Kommentar beim Link zur WorkoutList dagelassen :) Viel Spaß weiterhin!

**Pull request merged**
